src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/SlhDsaX509SignatureGenerator.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Formats.Asn1;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    internal sealed class SlhDsaX509SignatureGenerator : X509SignatureGenerator
+    {
+        private readonly SlhDsa _key;
+
+        internal SlhDsaX509SignatureGenerator(SlhDsa key)
+        {
+            Debug.Assert(key != null);
+
+            _key = key;
+        }
+
+        public override byte[] GetSignatureAlgorithmIdentifier(HashAlgorithmName hashAlgorithm)
+        {
+            // Ignore the hashAlgorithm parameter.
+            // This generator only supports SLH-DSA "Pure" signatures, but the overall design of
+            // CertificateRequest makes it easy for a hashAlgorithm value to get here.
+
+            const int InitialCapacity = 16;
+
+            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER, InitialCapacity);
+            using (writer.PushSequence())
+            {
+                writer.WriteObjectIdentifier(_key.Algorithm.Oid);
+            }
+
+            Debug.Assert(writer.GetEncodedLength() <= InitialCapacity);
+            return writer.Encode();
+        }
+
+        public override byte[] SignData(byte[] data, HashAlgorithmName hashAlgorithm)
+        {
+            ArgumentNullException.ThrowIfNull(data);
+
+            // Ignore the hashAlgorithm parameter.
+            // This generator only supports SLH-DSA "Pure" signatures, but the overall design of
+            // CertificateRequest makes it easy for a hashAlgorithm value to get here.
+
+            byte[] signature = new byte[_key.Algorithm.SignatureSizeInBytes];
+            int written = _key.SignData(data, signature);
```

Comment:
Seems weird to me that SlhDsa does not have a `byte[] SignData(byte[] data)` overload.

Reply:
Planning to do that in a future PR (along with the removal of the destination length return value in this overload).